test/goose/brokers/redis/batch_test.clj

```diff
@@ -0,0 +1,105 @@
+            _ ((batch/update-state (fn [_opts _job] "Function executed"))
+               opts job)
+            enqueued-job-set-size (redis-cmds/set-size (:redis-conn opts)
+                                                       enqueued-job-set)
+            successful-job-set-size (redis-cmds/set-size (:redis-conn opts)
+                                                         successful-job-set)]
+        (is (= 1 successful-job-set-size))
+        (is (= 0 enqueued-job-set-size))))
+    (testing "[redis] Job-id transition from enqueued -> dead"
+      (let [job (assoc job :retry-opts {:max-retries 0})
+            _ (redis-cmds/add-to-set (:redis-conn opts) enqueued-job-set job-id)
+            _ (try
+                ((batch/update-state (fn [_opts _job]
+                                       (throw (Exception. "Exception")))) opts job)
+                (catch Exception e
+                  (.getMessage e)))
+            enqueued-job-set-size (redis-cmds/set-size (:redis-conn opts)
+                                                       enqueued-job-set)
+            dead-job-set-size (redis-cmds/set-size (:redis-conn opts)
+                                                   dead-job-set)]
+        (is (= 0 enqueued-job-set-size))
+        (is (= 1 dead-job-set-size))))
+    (testing "[redis] Job-id transition from enqueued -> retrying"
+      (let [_ (redis-cmds/add-to-set (:redis-conn opts) enqueued-job-set job-id)
+            _ (try
+                ((batch/update-state (fn [_opts _job]
+                                       (throw (Exception. "Exception")))) opts job)
+                (catch Exception e
+                  (.getMessage e)))
+            enqueued-job-set-size (redis-cmds/set-size (:redis-conn opts)
+                                                       enqueued-job-set)
+            retrying-job-set-size (redis-cmds/set-size (:redis-conn opts)
+                                                       retry-job-set)]
+        (is (= 0 enqueued-job-set-size))
+        (is (= 1 retrying-job-set-size))))
+    (testing "[redis] Job-id transition from retrying -> successful"
+      (let [job (assoc job :state {:error           "Exception"
+                                   :last-retried-at (u/epoch-time-ms)
+                                   :first-failed-at (u/epoch-time-ms)
+                                   :retry-count     0})
+            _ (redis-cmds/add-to-set (:redis-conn opts) retry-job-set job-id)
+            _ ((batch/update-state (fn [_opts _job] "Function Executed"))
+               opts job)
+            retrying-job-set-size (redis-cmds/set-size (:redis-conn opts)
+                                                       retry-job-set)
+            successful-job-set-size (redis-cmds/set-size (:redis-conn opts)
+                                                         successful-job-set)]
+        (is (= 0 retrying-job-set-size))
```

Comment:
suggestion: I think its fine to just write this. Its a small enough statement, so executing this and assigning its value to another variable doesnt' seem like an advantage. Your test now becomes a set of setup statements and assertions
`(is (= 0 (redis-cmds/set-size (:redis-conn opts) retry-job-set))`